infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -477,6 +479,10 @@ protected void initChannel(@Nonnull Channel ch) throws Exception {
                     ch.pipeline().addLast("sasl/plain-text", new
                             PlainTextSaslNettyServer());
                 }
+
+                ch.pipeline().addLast(new Lz4FrameDecoder());
+                ch.pipeline().addLast(new Lz4FrameEncoder());
```

Comment:
The test is failing because the block size defaults to 64kb, so at the 64kb boundary the buffer is is compressed and sent even if the message is greater than 64kb this breaks the NettyCorfuMessageEncoder because it assumes the buffer is not segmented (i.e. contains the whole message).